### ISSUE TITLE
Feature request: Switch TableBlock’s header controls to a field that requires user input

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#5989]
### Is your proposal related to a problem?

Related: #5865, #5977. Wagtail’s [TableBlock](https://docs.wagtail.io/en/v2.8.1/reference/contrib/table_block.html#tableblock) has support for defining the first row, or column, as table headers. 

While it’s not mandatory for a table to have header cells, if the table isn’t meant for layout then using header cells is a usability and accessibility benefit (or requirement) – header cells are generally displayed differently (e.g. bold), and screen readers will use the header cell’s label when announcing items is the header for. I believe failing to mark up header cells as such is a failure of [WCAG2.1’s Success Criterion 1.3.1 (Info and Relationships)](https://www.w3.org/TR/WCAG20-TECHS/F91.html).

I think the current form makes it a bit too easy to miss those checkboxes and leave the table without headers.

### Describe the solution you'd like

I would suggest changing the header cells’ two checkboxes to a single "Header cells" dropdown, with the following options:

- [Choose] (Default)
- First row
- First column
- First row and first column
- No header cells

This field would be required. The main improvement here is the fact that contrary to checkboxes, the field has to be interacted with for the user to select the "No header cells" option that we want to discourage the use of (but should still be an option).

### Describe alternatives you've considered

One alternative would be to do this as a per-project customization, but it feels like an improvement that could easily live in Wagtail directly. Other changes to the TableBlock to remove the ability to have no header cells aren’t particularly compelling, because we still want the ability to select row/column/both.

### Additional context

Additionally to the above, as part of this change we could consider:

- Improving the help text for this field, to clarify its usage and the accessibility consequences
- Adding a straightforward way to customize this to disable the "No header cells" option – in practice all data tables should have header cells, and a table that doesn’t have any would be better as a list, or heading sections

In terms of migration, this looks like it could be easy:

- First row checkbox -> First row dropdown option
- First column checkbox -> First column dropdown option
- Both checkboxes -> "Both" dropdown option
- No checkboxes -> "No header cells" option

**Comment by thibaudcolas** [comment]
cc @zmustapha and @rjpruitt16 if you have any thoughts on this.

**Comment by gasman** [comment]
Just to add, I don't think it's feasible for us to perform a data migration on TableBlock data that already exists in people's projects. That doesn't rule out making the change at the UI level, though - we'd just need to make the appropriate translations between the `first_row_is_table_header` / `first_col_is_header` flags (as stored in the JSON) and the dropdown values, either within `render_form` / `value_from_datadict` or on the Javascript side.

**Comment by thibaudcolas** [comment]
👍 good point, that seems like a neat solution. And we won’t have to change the table’s front-end rendering, which already works as intended with the current setup.

**Comment by thibaudcolas** [comment]
Marking this as "good first issue", and removing "Needs Design Decision" – if anyone wants to pick this up please post a comment and go ahead. Changing this require some Django knowledge, and changing the UI to a `<select>`. Not super straightforward but should be approachable, and it should be relatively self-contained / easy to write tests for.

**Comment by ParitoshKabra** [comment]
This issue has two very old PR's active 😅 ? Can i still take up this issue @thibaudcolas @lb- ??

**Comment by thibaudcolas** [comment]
@ParitoshKabra as you’ve noticed this has already been assigned to someone and there is a PR awaiting review, so no.

**Comment by salty-ivy** [comment]
@cnk  shall i work on this issue?

**Comment by cnk** [comment]
@salty-ivy Yes please this is still something that could use some attention. Please look at gasman's comments about how to preserve compatibility with existing table data. And if you have any questions, please post them here so Thibaud or I can advise.

**Comment by salty-ivy** [comment]
okay, thanks a lot !

**Comment by ArijeetBanerjee** [comment]
I want to work on this issue. kindly give me permission and access to work on this issue

**Comment by DevilsAutumn** [comment]
Can i work on this issue?

**Comment by lb-** [comment]
@DevilsAutumn go for it, there is an existing PR (draft) with feedback. Please review that and I would suggest building a new PR based on where it started.

**Comment by DevilsAutumn** [comment]
Im trying to create (on main) and its not showing the + button which is used to add tables, while on v5.2.2 the same code is working fine. here is a ss:

<img width="996" alt="Screenshot 2023-12-09 at 2 43 41 PM" src="https://github.com/wagtail/wagtail/assets/83907321/81017903-3498-4608-b1b2-02082c650dba">
